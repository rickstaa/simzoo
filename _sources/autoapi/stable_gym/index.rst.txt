stable_gym
==========

.. py:module:: stable_gym

.. autoapi-nested-parse::

   Entry_point that register the Stable Gym gymnasium environments.



Subpackages
-----------

.. toctree::
   :maxdepth: 1

   /autoapi/stable_gym/common/index
   /autoapi/stable_gym/envs/index


Submodules
----------

.. toctree::
   :maxdepth: 1

   /autoapi/stable_gym/version/index


Attributes
----------

.. autoapisummary::

   stable_gym.__version__
   stable_gym.__version_tuple__
   stable_gym.ENVS


Classes
-------

.. autoapisummary::

   stable_gym.MaxEpisodeStepsInjectionWrapper


Package Contents
----------------

.. py:class:: MaxEpisodeStepsInjectionWrapper(env)

   Bases: :py:obj:`gymnasium.Wrapper`


   A gymnasium wrapper that injects the ``max_episode_steps`` attribute of the
   ``TimeLimit`` wrapper into the base environment as the
   ``_time_limit_max_episode_steps`` attribute. If the environment is not wrapped in
   a ``TimeLimit`` wrapper, then the ``_time_limit_max_episode_steps`` attribute is
   set to ``None``.

   Wrap a gymnasium environment.
   :param env: The gymnasium environment.
   :type env: gym.Env


   .. py:attribute:: max_episode_steps


.. py:data:: __version__
   :value: '0.20.7'


.. py:data:: __version_tuple__

.. py:data:: ENVS

