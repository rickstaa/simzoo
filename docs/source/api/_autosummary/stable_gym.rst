stable\_gym
===========

.. automodule:: stable_gym
  
   
   
   

   
   
   

   
   
   

   
   
   



.. rubric:: Modules

.. autosummary::
   :toctree:
   :template: custom-module-template.rst
   :recursive:

   stable_gym.common
   stable_gym.envs

