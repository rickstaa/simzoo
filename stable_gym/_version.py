# Stores the package version number so that it can be accessed from other modules.
version = "0.6.4"
version_tuple = version.split(".")
