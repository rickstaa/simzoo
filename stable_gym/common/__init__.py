"""Contains several functions that can be used across all the Stable Gym gymnasium
environments.
"""
from stable_gym.common.disturber import Disturber  # noqa: F401
