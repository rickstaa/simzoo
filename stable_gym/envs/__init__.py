"""The Stable Gym gymnasium environments."""
