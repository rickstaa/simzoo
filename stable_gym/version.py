# Stores the package version number so that it can be accessed from other modules.
__version__ = "0.6.38"
__version_tuple__ = __version__.split(".")
