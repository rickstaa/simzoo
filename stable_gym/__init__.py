"""Module that register the Stable Gym gymnasium environments."""
import gymnasium as gym
from gymnasium.envs.registration import register

# Make module version available.
from .version import __version__, __version_tuple__

# Available environments.
# TODO: Update reward thresholds.
ENVS = {
    "Oscillator-v1": {
        "module": "stable_gym.envs.biological.oscillator.oscillator:Oscillator",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "OscillatorComplicated-v1": {
        "module": "stable_gym.envs.biological.oscillator_complicated.oscillator_complicated:OscillatorComplicated",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "CartPoleCost-v1": {
        "module": "stable_gym.envs.classic_control.cartpole_cost.cartpole_cost:CartPoleCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "Ex3EKF-v1": {
        "module": "stable_gym.envs.classic_control.ex3_ekf.ex3_ekf:Ex3EKF",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "AntCost-v1": {
        "module": "stable_gym.envs.mujoco.ant_cost.ant_cost:AntCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "HalfCheetahCost-v1": {
        "module": "stable_gym.envs.mujoco.half_cheetah_cost.half_cheetah_cost:HalfCheetahCost",
        "max_step": 200,
        "reward_threshold": 300,
    },
    "HopperCost-v1": {
        "module": "stable_gym.envs.mujoco.hopper_cost.hopper_cost:HopperCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "SwimmerCost-v1": {
        "module": "stable_gym.envs.mujoco.swimmer_cost.swimmer_cost:SwimmerCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "Walker2dCost-v1": {
        "module": "stable_gym.envs.mujoco.walker2d_cost.walker2d_cost:Walker2dCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
}

for env, val in ENVS.items():
    if env not in gym.envs.registry:  # NOTE: Required because we use namespace packages
        register(
            id=env,
            entry_point=val["module"],
            max_episode_steps=val["max_step"],
            reward_threshold=val["reward_threshold"],
        )
