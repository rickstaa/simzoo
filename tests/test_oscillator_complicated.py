"""Test if the Oscillator environment still behaves like the previous Oscillator
environment.
"""
import gymnasium as gym
from gymnasium.logger import ERROR

import stable_gym  # noqa: F401

gym.logger.set_level(ERROR)


class TestOscillatorComplicated:
    env = gym.make(
        "OscillatorComplicated", exclude_reference_error_from_observation=False
    )

    def test_reset(self, snapshot):
        """Test if reset is still equal to the last snapshot."""
        observation, info = self.env.reset(seed=42)
        assert (observation == snapshot).all()
        assert info == snapshot

    def test_step(self, snapshot):
        """Test if steps behave the same."""
        # Perform steps and check if observations are equal.
        self.env.action_space.seed(42)
        for _ in range(5):
            action = self.env.action_space.sample()
            observation, reward, terminated, truncated, info = self.env.step(action)
            assert (observation == snapshot).all()
            assert reward == snapshot
            assert terminated == snapshot
            assert truncated == snapshot
            assert info == snapshot

    def test_constant_snapshot(self, snapshot):
        """Test if the 'CartPoleCost' environment with 'constant' reference is still
        equal to snapshot.
        """
        env_cost_reference_tracking = gym.make(
            "CartPoleCost",
            reference_type="constant",
        )
        observation, info = env_cost_reference_tracking.reset(seed=42)
        assert (observation == snapshot).all()
        assert info == snapshot
        env_cost_reference_tracking.action_space.seed(42)
        for _ in range(5):
            action = env_cost_reference_tracking.action_space.sample()
            (
                observation,
                reward,
                terminated,
                truncated,
                info,
            ) = env_cost_reference_tracking.step(action)
            assert (observation == snapshot).all()
            assert reward == snapshot
            assert terminated == snapshot
            assert truncated == snapshot
            assert info == snapshot
