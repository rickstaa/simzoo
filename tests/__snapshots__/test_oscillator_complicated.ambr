# serializer version: 1
# name: TestOscillatorComplicated.test_reset
  3.8697802427798167
# ---
# name: TestOscillatorComplicated.test_reset.1
  2.1943921987602617
# ---
# name: TestOscillatorComplicated.test_reset.10
  dict({
    'reference': 8.0,
    'reference_constraint_error': -19.52911326056175,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -7.529113260561752,
    'state_of_interest': 0.47088673943824766,
  })
# ---
# name: TestOscillatorComplicated.test_reset.2
  4.292989599556912
# ---
# name: TestOscillatorComplicated.test_reset.3
  3.4868401452968194
# ---
# name: TestOscillatorComplicated.test_reset.4
  0.47088673943824766
# ---
# name: TestOscillatorComplicated.test_reset.5
  4.878111758183779
# ---
# name: TestOscillatorComplicated.test_reset.6
  3.805698509951765
# ---
# name: TestOscillatorComplicated.test_reset.7
  3.930321526384769
# ---
# name: TestOscillatorComplicated.test_reset.8
  8.0
# ---
# name: TestOscillatorComplicated.test_reset.9
  -7.529113260561752
# ---
# name: TestOscillatorComplicated.test_step
  6.087455541244575
# ---
# name: TestOscillatorComplicated.test_step.1
  2.541687540355613
# ---
# name: TestOscillatorComplicated.test_step.10
  51.2380654736653
# ---
# name: TestOscillatorComplicated.test_step.11
  False
# ---
# name: TestOscillatorComplicated.test_step.12
  False
# ---
# name: TestOscillatorComplicated.test_step.13
  dict({
    'reference': 8.219875313546899,
    'reference_constraint_error': -18.938201626083277,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -7.158076939630176,
    'state_of_interest': 1.0617983739167234,
  })
# ---
# name: TestOscillatorComplicated.test_step.14
  1.139080931751879
# ---
# name: TestOscillatorComplicated.test_step.15
  7.643327224435085
# ---
# name: TestOscillatorComplicated.test_step.16
  8.770023692875586
# ---
# name: TestOscillatorComplicated.test_step.17
  7.149056015353484
# ---
# name: TestOscillatorComplicated.test_step.18
  1.972083358080852
# ---
# name: TestOscillatorComplicated.test_step.19
  5.047006142681629
# ---
# name: TestOscillatorComplicated.test_step.2
  7.256617066092945
# ---
# name: TestOscillatorComplicated.test_step.20
  5.169439569741611
# ---
# name: TestOscillatorComplicated.test_step.21
  4.798206919670843
# ---
# name: TestOscillatorComplicated.test_step.22
  8.439533636705194
# ---
# name: TestOscillatorComplicated.test_step.23
  -6.467450278624343
# ---
# name: TestOscillatorComplicated.test_step.24
  41.827913106478086
# ---
# name: TestOscillatorComplicated.test_step.25
  False
# ---
# name: TestOscillatorComplicated.test_step.26
  False
# ---
# name: TestOscillatorComplicated.test_step.27
  dict({
    'reference': 8.439533636705194,
    'reference_constraint_error': -18.02791664191915,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -6.467450278624343,
    'state_of_interest': 1.972083358080852,
  })
# ---
# name: TestOscillatorComplicated.test_step.28
  0.0
# ---
# name: TestOscillatorComplicated.test_step.29
  6.251511988275885
# ---
# name: TestOscillatorComplicated.test_step.3
  5.005962881903876
# ---
# name: TestOscillatorComplicated.test_step.30
  6.135240746499774
# ---
# name: TestOscillatorComplicated.test_step.31
  10.330570533462584
# ---
# name: TestOscillatorComplicated.test_step.32
  2.0360113056763014
# ---
# name: TestOscillatorComplicated.test_step.33
  5.967118130030345
# ---
# name: TestOscillatorComplicated.test_step.34
  6.262476986417208
# ---
# name: TestOscillatorComplicated.test_step.35
  5.65416346694715
# ---
# name: TestOscillatorComplicated.test_step.36
  8.6587581932296
# ---
# name: TestOscillatorComplicated.test_step.37
  -6.622746887553299
# ---
# name: TestOscillatorComplicated.test_step.38
  43.860776336596906
# ---
# name: TestOscillatorComplicated.test_step.39
  False
# ---
# name: TestOscillatorComplicated.test_step.4
  1.0617983739167234
# ---
# name: TestOscillatorComplicated.test_step.40
  False
# ---
# name: TestOscillatorComplicated.test_step.41
  dict({
    'reference': 8.6587581932296,
    'reference_constraint_error': -17.9639886943237,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -6.622746887553299,
    'state_of_interest': 2.0360113056763014,
  })
# ---
# name: TestOscillatorComplicated.test_step.42
  1.4871808075434318
# ---
# name: TestOscillatorComplicated.test_step.43
  8.789847065819913
# ---
# name: TestOscillatorComplicated.test_step.44
  4.63145230142228
# ---
# name: TestOscillatorComplicated.test_step.45
  5.989849036301869
# ---
# name: TestOscillatorComplicated.test_step.46
  1.9138506273357234
# ---
# name: TestOscillatorComplicated.test_step.47
  6.609332960352666
# ---
# name: TestOscillatorComplicated.test_step.48
  6.868366886672139
# ---
# name: TestOscillatorComplicated.test_step.49
  6.967804944284334
# ---
# name: TestOscillatorComplicated.test_step.5
  4.9365278044943945
# ---
# name: TestOscillatorComplicated.test_step.50
  8.877332634950129
# ---
# name: TestOscillatorComplicated.test_step.51
  -6.963482007614406
# ---
# name: TestOscillatorComplicated.test_step.52
  48.490081670369555
# ---
# name: TestOscillatorComplicated.test_step.53
  False
# ---
# name: TestOscillatorComplicated.test_step.54
  False
# ---
# name: TestOscillatorComplicated.test_step.55
  dict({
    'reference': 8.877332634950129,
    'reference_constraint_error': -18.086149372664277,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -6.963482007614406,
    'state_of_interest': 1.9138506273357234,
  })
# ---
# name: TestOscillatorComplicated.test_step.56
  1.8273701647910303
# ---
# name: TestOscillatorComplicated.test_step.57
  3.3647837737494335
# ---
# name: TestOscillatorComplicated.test_step.58
  7.202539228614338
# ---
# name: TestOscillatorComplicated.test_step.59
  6.381329798471056
# ---
# name: TestOscillatorComplicated.test_step.6
  4.264234935283765
# ---
# name: TestOscillatorComplicated.test_step.60
  2.0369685189025293
# ---
# name: TestOscillatorComplicated.test_step.61
  7.619148513262692
# ---
# name: TestOscillatorComplicated.test_step.62
  7.197297241699376
# ---
# name: TestOscillatorComplicated.test_step.63
  7.508112493435574
# ---
# name: TestOscillatorComplicated.test_step.64
  9.095041255281616
# ---
# name: TestOscillatorComplicated.test_step.65
  -7.058072736379087
# ---
# name: TestOscillatorComplicated.test_step.66
  49.81639075201777
# ---
# name: TestOscillatorComplicated.test_step.67
  False
# ---
# name: TestOscillatorComplicated.test_step.68
  False
# ---
# name: TestOscillatorComplicated.test_step.69
  dict({
    'reference': 9.095041255281616,
    'reference_constraint_error': -17.96303148109747,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -7.058072736379087,
    'state_of_interest': 2.0369685189025293,
  })
# ---
# name: TestOscillatorComplicated.test_step.7
  4.252396658049173
# ---
# name: TestOscillatorComplicated.test_step.8
  8.219875313546899
# ---
# name: TestOscillatorComplicated.test_step.9
  -7.158076939630176
# ---
